src/NuGet.Core/NuGet.ProjectModel/NjPackageSpecReader.cs

```diff
@@ -0,0 +1,1765 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.Packaging.Core;
+using NuGet.RuntimeModel;
+using NuGet.Versioning;
+
+namespace NuGet.ProjectModel
+{
+    [Obsolete]
```

Comment:
Why not deprecate JsonPackageSpecReader directly?

Reply:
One of the functions in that class uses the STJ implementation so we shouldn't be deprecating that right?

Reply:
It's not that we don't have to deprecate it, but we just shouldn't use it. 

My thinking was if we leavel JsonPackageSpecReader the footprint of this PR is a lot smaller. 

Maybe make it a partial class that has all the methods that split between old & new implementation, but right now it's just a lot of changes.

Reply:
[JsonPackageSpecReader is used in a bunch of places](https://github.com/search?q=repo%3ANuGet%2FNuGet.Client+JsonPackageSpecReader&type=code), including nuget.exe and pack (nuget.commands), so it's not clear to me what the suggestion is for those calls if the entire JsonPackageSpecReader class is marked as obsolete.

I think Nikolche's suggestion is to leave the Newtonsoft.Json spec reader code inside `JsonPackageSpecReader.cs`, and just mark all the methods as obsolete, rather than moving them into a new file, where git's diff doesn't tell us if any of the method bodies have changed.

Reply:
Oh okay, yea I thought about that. I put the NJ code back in the original file.

Reply:
And switched to partial class